.github/workflows/e2e.yml

```diff
@@ -27,34 +57,61 @@ jobs:
       DEFAULT_TESTS: ${{ steps.matrix-conditionals.outputs.DEFAULT_TESTS }}
       UPGRADE_TESTS: ${{ steps.matrix-conditionals.outputs.UPGRADE_TESTS }}
       UPGRADE_LIGHT_TESTS: ${{ steps.matrix-conditionals.outputs.UPGRADE_LIGHT_TESTS }}
+      UPGRADE_IMPORT_MAINNET_TESTS: ${{ steps.matrix-conditionals.outputs.UPGRADE_IMPORT_MAINNET_TESTS }}
       ADMIN_TESTS: ${{ steps.matrix-conditionals.outputs.ADMIN_TESTS }}
+      PERFORMANCE_TESTS: ${{ steps.matrix-conditionals.outputs.PERFORMANCE_TESTS }}
+      STATEFUL_DATA_TESTS: ${{ steps.matrix-conditionals.outputs.STATEFUL_DATA_TESTS }}
     steps:
-      # use cli rather than event context to avoid race conditions (label added after push)
+      # use api rather than event context to avoid race conditions (label added after push)
       - id: matrix-conditionals
-        run: |
-          if [[ ${{ github.event_name }} == 'pull_request' ]]; then
-            echo "DEFAULT_TESTS=true" >> $GITHUB_OUTPUT
-            labels=$(gh pr view -R ${{github.repository}} ${{github.event.pull_request.number}} --json labels -q '.labels[].name')
-            if [[ "$labels" == *"UPGRADE_TESTS"* ]]; then
-              echo "UPGRADE_TESTS=true" >> $GITHUB_OUTPUT
-            fi
-
-            if [[ "$labels" == *"UPGRADE_LIGHT_TESTS"* ]]; then
-              echo "UPGRADE_LIGHT_TESTS=true" >> $GITHUB_OUTPUT
-            fi
-
-            if [[ "$labels" == *"ADMIN_TESTS"* ]]; then
-              echo "ADMIN_TESTS=true" >> $GITHUB_OUTPUT
-            fi
-          elif [[ ${{ github.event_name }} == 'merge_group' ]]; then
-            echo "DEFAULT_TESTS=true" >> $GITHUB_OUTPUT
-          elif [[ ${{ github.event_name }} == 'push' && ${{ github.ref }} == 'refs/heads/develop' ]]; then
-            echo "DEFAULT_TESTS=true" >> $GITHUB_OUTPUT
-          elif [[ ${{ github.event_name }} == 'schedule' ]]; then
-            echo "UPGRADE_TESTS=true" >> $GITHUB_OUTPUT
-            echo "UPGRADE_LIGHT_TESTS=true" >> $GITHUB_OUTPUT
-            echo "ADMIN_TESTS=true" >> $GITHUB_OUTPUT
-          fi
+        uses: actions/github-script@v7
+        with:
+          script: |
+            console.log(context);
```

Comment:
is it possible to move this file to a separate file? https://github.com/actions/github-script?tab=readme-ov-file#run-a-separate-file

Reply:
Yes, but that would require a checkout of the repo which would slow it down a bit.